modules/db/src/test/java/org/mule/module/db/integration/select/SelectsLongTransactionTestCase.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.module.db.integration.select;
+
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+import org.mule.api.MuleEventContext;
+import org.mule.api.MuleMessage;
+import org.mule.api.client.LocalMuleClient;
+import org.mule.api.lifecycle.Callable;
+import org.mule.module.db.integration.AbstractDbIntegrationTestCase;
+import org.mule.module.db.integration.TestDbConfig;
+import org.mule.module.db.integration.model.AbstractTestDatabase;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+public class SelectsLongTransactionTestCase extends AbstractDbIntegrationTestCase
+{
+
+    public SelectsLongTransactionTestCase(String dataSourceConfigResource, AbstractTestDatabase testDatabase)
+    {
+        super(dataSourceConfigResource, testDatabase);
+    }
+
+    @Parameterized.Parameters
+    public static List<Object[]> parameters()
+    {
+        return TestDbConfig.getResources();
+    }
+
+    @Override
+    protected String[] getFlowConfigurationResources()
+    {
+        return new String[] {"integration/select/selects-long-transaction-config.xml"};
+    }
+
+    @Test
+    public void oneWay() throws Exception
+    {
+        LocalMuleClient client = muleContext.getClient();
+
+        client.dispatch("vm://testOneWay", TEST_MESSAGE, null);
+        MuleMessage response = client.request("vm://testOut", RECEIVE_TIMEOUT);
+
+        assertThat(response.getExceptionPayload(), nullValue());
+    }
+
+    @Test
+    public void requestResponse() throws Exception
+    {
+        LocalMuleClient client = muleContext.getClient();
+
+        MuleMessage response = client.send("vm://testRequestResponse", TEST_MESSAGE, null);
+
+        assertThat(response.getExceptionPayload(), nullValue());
+    }
+
+    public static class SetCollectionFlowVar implements Callable
```

Comment:
Instead of using a component, you can send the collection as the message's payload in the test message